src/dvc_data/hashfile/hash.py

```diff
@@ -191,15 +197,22 @@ def hash_file(
 ) -> tuple["Meta", "HashInfo"]:
     if state:
         meta, hash_info = state.get(path, fs, info=info)
-        if hash_info and hash_info.name == name:
+        if meta is not None and hash_info is not None and hash_info.name == name:
             return meta, hash_info
 
-    cb = callback or LargeFileHashingCallback(desc=path)
-    with cb:
-        hash_value, meta = _hash_file(path, fs, name, callback=cb, info=info)
-    hash_info = HashInfo(name, hash_value)
+    size = info.get("size") if info else None
+    _callback = callback
+    # never initialize callback if it's never going to be used
+    if size and size < LargeFileHashingCallback.LARGE_FILE_SIZE:
+        _callback = nullcontext(None)
+    else:
+        _callback = LargeFileHashingCallback(desc=path)
```

Comment:
If the file size is <1GB, we don't show progress bars. But only the `tqdm` was not being called in that case. `fsspec`'s callback was still called, where we'd waste some time. 

With the current change, we won't create callback at all if the file size is <1GB.